google-apis-core/lib/google/apis/core/storage_download.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'google/apis/core/api_command'
+require 'google/apis/errors'
+require 'addressable/uri'
+
+module Google
+  module Apis
+    module Core
+      # Streaming/resumable media download support specifically for storage API so that 
+      # we can respond with response headers too.
+      class StorageDownloadCommand < DownloadCommand
+
+        # Execute the upload request once. Overrides the default implementation to handle streaming/chunking
+        # of file content.
+        #
+        # @private
+        # @param [HTTPClient] client
+        #   HTTP client
+        # @yield [result, err] Result or error if block supplied
+        # @return [Object]
+        # @raise [Google::Apis::ServerError] An error occurred on the server and the request can be retried
+        # @raise [Google::Apis::ClientError] The request is invalid and should not be retried without modification
+        # @raise [Google::Apis::AuthorizationError] Authorization is required
+        def execute_once(client, &block)
+          request_header = header.dup
```

Comment:
I haven't deliberately refactored the code and let it be redundant as we didn't want to spend too much time doing that.